DFE-Digital/sts-plan-technology-for-your-school
# Feat: Simplify headers on recommendation chunks

## Overview

Addresses ticket [#226338](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_workitems/edit/226338)

Changes RecommendationChunks to have a short text field for Headers instead of a separate Header content type. This is to simplify the creation process for content designers

## Changes

### Major

- RecommendationChunk no longer has a linked Header, but a text field for the Header
- Creates the groundwork for doing contentful migrations programatically, manual at the moment

### Minor

- Added some content to the seeded test db for running the database script and testing this locally
- Changes IHeaderWithContent to have a `HeaderText` field instead of `Header` so that both `RecommendationIntro` with a separate header, and `RecommendationChunk` with a text base one can inherit from it
  - I considered a few different ways to go about this, such as having only the intro inherit from it, and treating chunks and intros separately, but it didn't work particularly well so I think this is the better of the options

## How to review the PR

The column has already been added to the database in dev (but the id column not removed) so this should run locally already 

The seeded test database can also be used to check that the change is working, headers have been added of the new style to the connectivity test category

The migration script can be tested before its run across dev by
- Go to contentful
- Duplicate the Recommendation Chunk content type
- Update the migration script and change `content_type` to your test content type
- Add some chunks with your duplicated type
- Run the migration script following the readme instructions
- Verify that your chunks of the duplicated type have swapped linked headers for text headers

## Release requirements

The contentful migration script will need to be run manually just after the code merges.
This will transform all the recommendation chunks, but it won't delete the old headers. the headers also need removing

I'm looking into the best solution to that. The contentful-migration cli does not support deleting entries but the contentful-management one does. They don't go together particularly well so this may have to be a separate setup. 

## Checklist
